### PR TITLE
Apply changes to main

### PR DESCRIPTION
**Pull request by 0xcodercrane**
Resolves #

### CODE REVIEW

**Comment by pavlovcik** [comment]
QA is the most important part here. @Draeieg you might need to spend a day on this. 

I wonder if it makes more sense for you to only do QA on these syncs to `main` vs on every pull request? 

Alternatively it would make sense for you to immediate revert pull requests if they don't pass QA so that those don't make it into these syncs, making your QA unnecessary for these syncs.

File: .github/ubiquibot-config.yml [hunk 1]
```diff
@@ -28,5 +29,5 @@ priority-labels:
   - name: "Priority: 4 (Emergency)"
     weight: 5
 auto-pay-mode: true
-analytics-mode: true
+analytics-mode: false
```
Comment (pavlovcik) [inline .github/ubiquibot-config.yml:32]: Why

File: bin/action.yml
Comment (pavlovcik) [inline bin/action.yml:1]: Is this needed

File: package.json [hunk 2]
```diff
@@ -13,8 +13,10 @@
   "scripts": {
     "build": "tsc",
     "build:serverless": "ncc build src/index.ts -o dist",
+    "build:ci": "ncc build adapters/index.ts -o dist && mv ./dist/index.js ./bin",
```
Comment (pavlovcik) [inline package.json:16]: We should convert everything fully into typescript.

File: src/handlers/comment/handlers/multiplier.ts [hunk 3]
```diff
@@ -18,13 +18,37 @@ export const multiplier = async (body: string) => {
     return;
   }
 
-  const regex = /\/multiplier @(\w+) (\d+(?:\.\d+)?)/; // /multiplier @0xcodercrane 0.5
+  const regex = /(".*?"|[^"\s]+)(?=\s*|\s*$)/g;
+  /** You can use this command to set a multiplier for a user.
```
Comment (pavlovcik) [inline src/handlers/comment/handlers/multiplier.ts:22]: Shouldn't this regular expression begin with `/multiplier`?

File: src/handlers/comment/handlers/payout.ts [hunk 4]
```diff
@@ -27,7 +27,7 @@ export const payout = async (body: string) => {
   }
 
   const IssueComments = await getAllIssueComments(issue.number);
-  if (IssueComments.length === 0) {
+  if (IssueComments.length == 0) {
```
Comment (pavlovcik) [inline src/handlers/comment/handlers/payout.ts:30]: Why

File: src/handlers/push/index.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  // skip if empty
+  if (payload.head_commit && payload.head_commit.modified.length === 0 && payload.head_commit.added.length === 0) {
+    logger.debug("Skipping push events, file change empty");
+    return;
+  }
+
+  // check for modified or added files and check for specified file
+  if (payload.head_commit.modified.includes(BASE_RATE_FILE) || payload.head_commit.added.includes(BASE_RATE_FILE)) {
+    // update base rate
+    await updateBaseRate(context, payload, BASE_RATE_FILE);
```
Comment (pavlovcik) [inline src/handlers/push/index.ts:29]: Shouldn't we be updating all the price labels here too?

File: src/handlers/push/update-base.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+  // fetch all labels
+  const repoLabels = await listLabelsForRepo();
+
+  if (repoLabels.length === 0) {
+    logger.debug("No labels on this repo");
+    return;
+  }
+
+  await updateLabelsFromBaseRate(owner, repo, context, repoLabels as Label[], previousBaseRate);
```
Comment (pavlovcik) [inline src/handlers/push/update-base.ts:35]: Or is this one handling my previous comment?

File: src/helpers/label.ts [hunk 7]
```diff
@@ -14,13 +17,13 @@ export const listLabelsForRepo = async (per_page?: number, page?: number): Promi
   });
 
   if (res.status === 200) {
-    return res.data.map((i) => i.name);
+    return res.data;
```
Comment (pavlovcik) [inline src/helpers/label.ts:20]: Why

**Comment by pavlovcik** [comment]
Maybe we can offer a job to wannacfuture and vennox to help with our reviews/qa/deploys? @Draeieg

File: bin/action.yml
Comment (0xcodercrane) [reply bin/action.yml:1]: I guess no need

Comment (0xcodercrane) [reply bin/action.yml:1]: https://github.com/ubiquity/ubiquibot/pull/422/commits/699cc1d740e98393294af4d57e3d0b5e47300306

File: .github/ubiquibot-config.yml (hunk 1 above)
Comment (0xcodercrane) [reply .github/ubiquibot-config.yml:32]: It is the default value of analytics mode. I am not sure why it has been changed. Let me revert it

Comment (0xcodercrane) [reply .github/ubiquibot-config.yml]: https://github.com/ubiquity/ubiquibot/pull/422/commits/55a0b32879330c281badf12dc4cb9ff9049d6151

File: package.json (hunk 2 above)
Comment (0xcodercrane) [reply package.json]: removed both `build:ci` and `build:serverless`

File: src/handlers/comment/handlers/multiplier.ts (hunk 3 above)
Comment (0xcodercrane) [reply src/handlers/comment/handlers/multiplier.ts:22]: This regex looks like a more global one. we also have a command type check [here](https://github.com/ubiquity/ubiquibot/blob/development/src/handlers/comment/handlers/index.ts#L34) so i guess its fine to keep

File: src/handlers/comment/handlers/payout.ts (hunk 4 above)
Comment (0xcodercrane) [reply src/handlers/comment/handlers/payout.ts:30]: same thing. no need to use `==` instead of `===`

File: bin/index.js
Comment (pavlovcik) [inline bin/index.js:1]: This should not be committed to the repo. @rndquu proposed adding a `.gitkeep` file to track this directory instead.

File: .github/workflows/build.yml [hunk 8]
```diff
@@ -21,3 +21,5 @@ jobs:
       - name: Local Build
         run: yarn build
 
+      - name: Lint
+        run: yarn lint
```
Comment (pavlovcik) [inline .github/workflows/build.yml:25]: Why bother linting when building it just slows it down. 

```suggestion
```

File: src/adapters/supabase/helpers/client.ts [hunk 9]
```diff
@@ -64,7 +64,7 @@ export type IssueAdditions = {
   completed_at?: number;
 };
 
-const getDbDataFromIssue = (issue: Issue, additions: IssueAdditions): any => {
+const getDbDataFromIssue = (issue: Issue, additions: IssueAdditions) => {
   return {
     issue_number: issue.number,
     issue_url: issue.html_url,
```
Comment (pavlovcik) [inline src/adapters/supabase/helpers/client.ts:70]: What is the purpose of getting the issue and url from the database when we're working directly on GitHub on issues.

File: .eslintrc.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+  parserOptions: {
+    tsconfigRootDir: __dirname,
+    project: ["./tsconfig.json"],
+  },
+  plugins: ["@typescript-eslint"],
+  extends: ["eslint:recommended", "plugin:@typescript-eslint/recommended"],
+  rules: {
+    "@typescript-eslint/no-floating-promises": "error",
+  },
+};
```
[the remark this answers is not in the document]
Comment (0xcodercrane) [reply .eslintrc.js]: https://github.com/ubiquity/ubiquibot/pull/422/commits/13558daa004cff1ce3a454478783e5ff34a704fb

File: src/helpers/label.ts (hunk 7 above)
Comment (0xcodercrane) [reply src/helpers/label.ts:20]: I guess someone wanted to use all the info of labels

File: bin/index.js
Comment (0xcodercrane) [reply bin/index.js:1]: I might be missing @rndquu 's suggestion. By the way, `.gitkeep` file is used to track empty directories. so we need to keep the empty bin directory. I don't think we will use it in the near future.

**Comment by pavlovcik** [comment]
@Draeieg when you can start QA?

**Comment by Draeieg** [comment]
> @Draeieg when you can start QA?

I have everything ready to test now. just need the credentials for env variables and logs

**Comment by pavlovcik** [comment]
I'll send you them now.

**Review by pavlovcik — APPROVED** [review]

**Comment by 0xcodercrane** [comment]
Please approve if we're good to go @Draeieg

**Comment by Draeieg** [comment]
> Please approve if we're good to go @Draeieg 

We should have a list of added features in this build so I can approve easier 

Currently, we have some broken stuff on staging, so we need to be 100% sure those won't go through

**Comment by 0xcodercrane** [comment]
> > Please approve if we're good to go @Draeieg
> 
> We should have a list of added features in this build so I can approve easier
> 
> Currently, we have some broken stuff on staging, so we need to be 100% sure those won't go through
We should pick up the merge commits from the ones the PR has. If the PR has tons of commits in it, its kind of tedious thing to filter by the merge commits.

To make it easier, I am still suggesting to squash and merge individual PRs(originated from hunters fork)   @pavlovcik

**Comment by pavlovcik** [comment]
> To make it easier, I am still suggesting to squash and merge individual PRs(originated from hunters fork) @pavlovcik.

We'll lose the commit data. I'm not a fan of losing data. And branches automatically should be deleted or else it gets unwieldily after a short while.

**Comment by 0xcodercrane** [comment]
This has been merged into main mistakenly. so reverted. 🙏

**Pull request merged**